### PR TITLE
ingest: Add support for CAP-63

### PR DESCRIPTION
**Pull request by tamirms**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've reviewed the changes in this PR and if I consider them worthwhile for being mentioned on release notes then I have updated the relevant `CHANGELOG.md` within the  component folder structure. For example, if I changed horizon, then I updated ([services/horizon/CHANGELOG.md](services/horizon/CHANGELOG.md). I add a new line item describing the change and reference to this PR. If I don't update a CHANGELOG, I acknowledge this PR's change may not be mentioned in future release notes.  
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Fixes https://github.com/stellar/go/issues/5614

This PR implements support for CAP-63 see https://github.com/stellar/go/issues/5614 for more details.

The failing tests are fixed in https://github.com/stellar/go/pull/5611 . Once https://github.com/stellar/go/pull/5611 is merged , I will rebase this PR and the tests should pass.
### Known limitations

[N/A]

### CODE REVIEW

File: ingest/change_compactor.go [hunk 1]
```diff
@@ -163,6 +163,13 @@ func (c *ChangeCompactor) getLedgerKey(ledgerEntry *xdr.LedgerEntry) ([]byte, er
 	return ledgerKey, nil
 }
 
+func compactTTLUpdate(a, b xdr.TtlEntry) xdr.TtlEntry {
```
Comment (sreuland) [inline ingest/change_compactor.go:166]: could the parameter names be more descriptive, would help with context on what this is doing.

File: ingest/change_compactor.go [hunk 2]
```diff
@@ -179,22 +186,19 @@ func (c *ChangeCompactor) addUpdatedChange(change Change) error {
...
-	case xdr.LedgerEntryChangeTypeLedgerEntryCreated:
-		// If existing type is created it means that this entry does not
-		// exist in a DB so we update entry change.
-		c.cache[ledgerKeyString] = Change{
-			Type:       change.Type,
-			Pre:        existingChange.Pre, // = nil
-			Post:       change.Post,
-			ChangeType: existingChange.ChangeType,
+	case xdr.LedgerEntryChangeTypeLedgerEntryCreated,
+		xdr.LedgerEntryChangeTypeLedgerEntryUpdated,
```
Comment (sreuland) [inline ingest/change_compactor.go:190]: LedgerEntryUpdated's are now cached into the compactor all the time, before it was dropped, don't see any changes in existing tests that may have broken related to this, I do see the new coverage on TTL's, is there potential edge case where LedgerEntryUpdated should still be dropped or are TTL's the only source of an LedgerEntryUpdated?

File: xdr/ledger_close_meta.go [hunk 3]
```diff
@@ -58,8 +58,21 @@ func (l LedgerCloseMeta) TransactionEnvelopes() []TransactionEnvelope {
 	case 1:
 		var envelopes = make([]TransactionEnvelope, 0, l.CountTransactions())
 		for _, phase := range l.MustV1().TxSet.V1TxSet.Phases {
-			for _, component := range *phase.V0Components {
-				envelopes = append(envelopes, component.TxsMaybeDiscountedFee.Txs...)
+			switch phase.V {
```
Comment (sreuland) [inline xdr/ledger_close_meta.go:61]: should the new v1, and panic cases be exercised through unit tests? don't see any coverage directly against this method, or could those branches be asserted indirectly through LedgerTransactionReader tests or other?

**Review by sreuland — APPROVED** [review]
lgtm, left comments for consideration on test coverage.

File: ingest/change_compactor.go (hunk 2 above)
Comment (tamirms) [reply ingest/change_compactor.go:190]: > LedgerEntryUpdated's are now cached into the compactor all the time, before it was dropped

the behavior is actually the same. The [fallthrough](https://go.dev/wiki/Switch#fall-through) keyword means it should match the case below. In other words,

```go
	case xdr.LedgerEntryChangeTypeLedgerEntryUpdated:
		fallthrough
	case xdr.LedgerEntryChangeTypeLedgerEntryRestored:
```

is equivalent to:

```go
	case xdr.LedgerEntryChangeTypeLedgerEntryUpdated, xdr.LedgerEntryChangeTypeLedgerEntryRestored:
```

File: ingest/change_compactor.go (hunk 1 above)
Comment (tamirms) [reply ingest/change_compactor.go]: renamed the function and added a comment to make it more clear

File: xdr/ledger_close_meta.go (hunk 3 above)
Comment (tamirms) [reply xdr/ledger_close_meta.go:61]: I added a test case

File: xdr/ledger_close_meta.go [hunk 4]
```diff
@@ -58,8 +58,21 @@ func (l LedgerCloseMeta) TransactionEnvelopes() []TransactionEnvelope {
...
+					envelopes = append(envelopes, component.TxsMaybeDiscountedFee.Txs...)
+				}
+			case 1:
+				for _, stage := range phase.ParallelTxsComponent.ExecutionStages {
+					for _, cluster := range stage {
+						for _, envelope := range cluster {
+							envelopes = append(envelopes, envelope)
+						}
+					}
+				}
```
[the remark this answers is not in the document]
Comment (chowbao) [reply xdr/ledger_close_meta.go:73]: Yeah we will make the changes in stellar-etl. TBH I'm not sure exactly where the logic is used in stellar-etl but a quick grep will fix that when we make the update
